File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/servers/library/views/handlers/OnGetLibraryViewItemResultsComplete.java
package com.lasthopesoftware.bluewater.servers.library.views.handlers;

import android.app.Activity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;

import com.lasthopesoftware.bluewater.servers.library.items.Item;
import com.lasthopesoftware.bluewater.servers.library.items.list.ClickItemListener;
import com.lasthopesoftware.bluewater.servers.library.items.list.menus.changes.handlers.IItemListMenuChangeHandler;
import com.lasthopesoftware.threading.ISimpleTask;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by david on 11/5/15.
 */
public class OnGetLibraryViewItemResultsComplete extends OnGetLibraryViewIItemResultsComplete<Item> {
    private final ListView listView;
    private final Activity activity;

    public OnGetLibraryViewItemResultsComplete(Activity activity, ViewGroup container, ListView listView, View loadingView, int position, IItemListMenuChangeHandler itemListMenuChangeHandler) {
        super(activity, container, listView, loadingView, position, itemListMenuChangeHandler);

        this.listView = listView;
        this.activity = activity;
    }

    public void onComplete(ISimpleTask<Void, Void, List<Item>> owner, List<Item> result) {
        super.onComplete(owner, result);
        if (result == null) return;

        listView.setOnItemClickListener(new ClickItemListener(activity, result instanceof ArrayList ? (ArrayList<Item>) result : new ArrayList<>(result)));
    }
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/views/handlers/OnGetLibraryViewItemResultsComplete.kt
package com.lasthopesoftware.bluewater.client.browsing.library.views.handlers

import android.app.Activity
import android.view.View
import android.widget.ListView
import com.lasthopesoftware.bluewater.R
import com.lasthopesoftware.bluewater.client.browsing.items.Item
import com.lasthopesoftware.bluewater.client.browsing.items.list.ClickItemListener
import com.lasthopesoftware.bluewater.client.browsing.items.list.DemoableItemListAdapter
import com.lasthopesoftware.bluewater.client.browsing.items.list.menus.changes.handlers.IItemListMenuChangeHandler
import com.lasthopesoftware.bluewater.client.browsing.items.media.files.access.parameters.IFileListParameterProvider
import com.lasthopesoftware.bluewater.client.browsing.items.menu.LongClickViewAnimatorListener
import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library
import com.lasthopesoftware.bluewater.client.stored.library.items.StoredItemAccess
import com.namehillsoftware.handoff.promises.response.ImmediateResponse

/**
 * Created by david on 11/5/15.
 */
class OnGetLibraryViewItemResultsComplete(private val activity: Activity, private val listView: ListView, private val loadingView: View, private val itemListMenuChangeHandler: IItemListMenuChangeHandler, private val fileListParameterProvider: IFileListParameterProvider, private val storedItemAccess: StoredItemAccess, private val library: Library) : ImmediateResponse<List<Item>, Unit> {
	override fun respond(result: List<Item>) {
		listView.onItemLongClickListener = LongClickViewAnimatorListener()
		listView.adapter = DemoableItemListAdapter(activity, R.id.tvStandard, result, fileListParameterProvider, itemListMenuChangeHandler, storedItemAccess, library)
		loadingView.visibility = View.INVISIBLE
		listView.visibility = View.VISIBLE
		listView.onItemClickListener = ClickItemListener(result, loadingView)
	}
}
